src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -19,24 +25,49 @@ public class FindCommand extends Command {
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
             + "Example: " + COMMAND_WORD + " gravity constant";
 
-    private final QuestionContainsKeywordsPredicate predicate;
+    private final QuestionContainsKeywordsPredicate cardPredicate;
+    private final DeckContainsKeywordsPredicate deckPredicate;
 
-    public FindCommand(QuestionContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    /**
+     * Constructs a FindCommand with the given list of keywords.
+     * The command will search for cards and decks that contain any of the keywords.
+     * @param keywords the list of keywords to search for
+     */
+    public FindCommand(List keywords) {
+        this.cardPredicate = new QuestionContainsKeywordsPredicate(keywords);
+        this.deckPredicate = new DeckContainsKeywordsPredicate(keywords);
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredCardList(predicate);
+        String returnMessage;
+        int returnListSize;
+        ModelState currentState = model.getState();
+        switch (currentState) {
+        case DECK_MODE:
+            model.updateFilteredCardList(cardPredicate);
```

Comment:
Interesting behaviour, but I wonder if its confusing or counterintuitive for user - What if they are in deck mode and they want to filter the decks too?

Reply:
hmm but deck mode means you have a deck selected right. I think it makes sense to be filtering for cards after a deck is selected?

Reply:
I thought so because right now we allow users to select a deck from Deck mode too. Actually whats the point of showing the cards in the main mode anyway ahh, its not like we can interact with them right?

Reply:
We are currently displaying all cards when in main mode if I'm not wrong. I agree. I think we shouldn't show all cards in the main mode as we may confuse the user as to which mode they are in.